### PR TITLE
Fix division by zero and cast of too big floats to int

### PR DESCRIPTION
**Pull request by LukasWoodtli**
Not sure that this is the best way to fix that. But it seems to work.

There is some duplicated code that I would like to extract (for example into the math library). But I'm not sure if `float.h` is available on all platforms.

### CODE REVIEW

**Comment by dagar** [comment]
Yes float.h is available everywhere. 
Put this somewhere in mathlib? https://github.com/PX4/Firmware/tree/master/src/lib/mathlib/math

A float equality helper is something I've considered adding before. https://www.floating-point-gui.de/errors/comparison/

File: src/modules/simulator/gyrosim/gyrosim.cpp [hunk 1]
```diff
@@ -109,6 +109,20 @@ using namespace DriverFramework;
 #define EXTERNAL_BUS 0
 #endif
 
+namespace
+{
+/** Save way to check if float is zero */
+inline bool isZero(const float &val)
+{
+	return abs(val - 0.0f) < FLT_EPSILON;
```
Comment (dagar) [inline src/modules/simulator/gyrosim/gyrosim.cpp]: fabsf?

File: src/modules/simulator/gyrosim/gyrosim.cpp [hunk 2]
```diff
@@ -109,6 +109,20 @@ using namespace DriverFramework;
...
+}
+
+inline int16_t constrainToInt(float value)
```
Comment (dagar) [inline src/modules/simulator/gyrosim/gyrosim.cpp]: Mention the specific size?

File: src/modules/simulator/gyrosim/gyrosim.cpp (hunk 1 above)
Comment (LukasWoodtli) [reply src/modules/simulator/gyrosim/gyrosim.cpp]: abs works also for floating point number with a C++ sdlib. But we don’t have it on Nuttx (IIRC)? I’ll make it C compliant.

**Comment by LukasWoodtli** [comment]
A general float equality helper could be added but comparison with zero is a special case that needs to be handled separately (it seems that Java does this differently).

**Pull request merged**